launcher/container_runner_test.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/google/go-tpm-tools/cel"
+)
+
+// Fake attestation agent.
+type fakeAttestationAgent struct {
+	measureEventFunc func(cel.Content) error
+	attestFunc       func(context.Context) ([]byte, error)
+}
+
+func (f *fakeAttestationAgent) MeasureEvent(event cel.Content) error {
+	if f.measureEventFunc != nil {
+		return f.measureEventFunc(event)
+	}
+
+	return fmt.Errorf("unimplemented")
+}
+
+func (f *fakeAttestationAgent) Attest(ctx context.Context) ([]byte, error) {
+	if f.attestFunc != nil {
+		return f.attestFunc(ctx)
+	}
+
+	return nil, fmt.Errorf("unimplemented")
+}
+
+func createJWTToken(t *testing.T, ttl time.Duration) []byte {
+	t.Helper()
+
+	privkey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatalf("Error creating token key: %v", err)
+	}
+
+	now := jwt.TimeFunc().Unix()
+	claims := &jwt.StandardClaims{
+		Id:        "test token",
+		IssuedAt:  now,
+		NotBefore: now,
+		ExpiresAt: now + int64(ttl.Seconds()),
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	signed, err := token.SignedString(privkey)
+	if err != nil {
+		t.Fatalf("Error creating signed string: %v", err)
+	}
+
+	return []byte(signed)
+}
+
+func TestGetTTL(t *testing.T) {
+	expectedTTL := 5 * time.Second
+	token := createJWTToken(t, expectedTTL)
+
+	ttl, err := getTTL(token)
+	if err != nil {
+		t.Fatalf("getTTL failed: %v", err)
+	}
+
+	// Expect TTL to be greater than 0 and no greater than the expected TTL.
+	if ttl <= 0 {
+		t.Errorf("expect TTL to be greater than 0, got %v", ttl)
+	}
+
+	if ttl > expectedTTL {
+		t.Errorf("TTL exceeds the expected TTL: got %v, want no greater than %v", ttl, expectedTTL)
+	}
+}
+
+func TestGetTTLError(t *testing.T) {
+	_, err := getTTL([]byte("not a valid token"))
```

Comment:
can we test a TTL that's negative?

Reply:
Can you elaborate on what you mean here? Is it creating a token whose ExpiresAt has already passed?

Reply:
Yes, sorry for the confusion. ExpiresAt set to `time.Now() - 1`